src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java

```diff
@@ -452,7 +470,30 @@ public Result onBuildCompletedNew(ProgrammingExerciseParticipation participation
             final var hasArtifact = buildResult.getBuild().isArtifact();
             programmingSubmission.setBuildArtifact(hasArtifact);
             programmingSubmission.setBuildFailed(result.getResultString().equals("No tests found"));
-            programmingSubmissionRepository.save(programmingSubmission);
+
+            programmingSubmission = programmingSubmissionRepository.save(programmingSubmission);
+
+            Optional<ProgrammingSubmission> optionalProgrammingSubmission = programmingSubmissionRepository.findWithEagerBuildLogEntriesById(programmingSubmission.getId());
```

Comment:
Why do you fetch the programming submission with build logs here? Could you just use the `programmingSubmission` from a line above?

Reply:
Good catch! I first thought that, in order to set the build logs, they have to be present (i.e. not be a proxy), but as we are only setting them (and not getting), it is not required. I've changed it.